pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go

```diff
@@ -194,12 +178,38 @@ func (r *ReconcilePerconaServerMongoDBRestore) reconcileRestore(cr *psmdbv1.Perc
 	defer pbmc.Close()
 
 	if status.State == psmdbv1.RestoreStateNew || status.State == psmdbv1.RestoreStateWaiting {
-		stg, ok := cluster.Spec.Backup.Storages[storageName]
+		var (
+			backupName  = cr.Spec.BackupName
+			storageName = cr.Spec.StorageName
+		)
+
+		if cr.Spec.PITR != nil && !cluster.Spec.Backup.IsEnabledPITR() && backupName == "" && storageName == "" {
+			return errors.New("backup/storage name must be specified for point in time restore with pitr disabled in cluster")
+		}
+
+		if cr.Spec.PITR != nil && cluster.Spec.Backup.IsEnabledPITR() {
+			// getting the first and only storage if PITR is enabled
+			for storageName = range cluster.Spec.Backup.Storages {
+			}
+		} else if backupName == "" || storageName == "" {
+			bcp, err := r.getBackup(cr)
+			if err != nil {
+				return errors.Wrap(err, "get backup")
+			}
+			if bcp.Status.State != psmdbv1.BackupStateReady {
+				return errors.New("backup is not ready")
+			}
```

Comment:
why we moved this block?
this block was earlier in code because to not initiate the connection to PBM if we are not going to run any restoration

Reply:
fixed in f773c614c0f44980eb50d5bbd0aeefbe98af0531